src/dashboard/luasrc/controller/ids-priv/ids_tab.lua

```diff
@@ -22,11 +21,11 @@ module("luci.controller.ids-priv.ids_tab", package.seeall)  --notice that new_ta
         local payload = luci.jsonc.parse(tostring(config))
 
         -- Open configuration file as JSON
-        local conf = io.open("/etc/capstone-ids/config.json", 'r')
+        local conf = io.open("/etc/tinyHIPPO/config.json", 'r')
```

Comment:
In the python portion of our package we created a config object to avoid having to read from the file everytime a config value needs to be accessed. Is there a similar mechanism for Luci? If so can it be implemented here and in other parts of your luci code?

Reply:
Well the config file is being directly altered in order to write the new MAC's to the file, so it kinda requires an I/O operation. This is also the sole place where the config file is being accessed so creating another object just to track use in one place seems redundant.

Reply:
If that is the case, once we have SQLite in-place this will be removed right? Additionally can the I/O operation done on the file below (line 42) be removed and can this declaration be modified to open the file for reading and writing?

Reply:
Sadly, Lua doesn't work like that. The closest file operation to an "update" (Read and write) is a combo of read and append ("r+"). So in order to rewrite the MAC values inline (Without appending the whole thing again), I need to read and write in two operations. But yes, eventually this should be migrated once we move over the python config file.